lib/collective.lib.js

```diff
@@ -75,3 +76,43 @@ export const suggestSlug = value => {
 
   return trim(slugify(value, slugOptions), '-');
 };
+
+const getTopContributors = contributors => {
+  const topOrgs = [];
+  const topIndividuals = [];
+
+  for (const contributor of contributors) {
+    // We only care about financial contributors that donated $$$
+    if (!contributor.isBacker || !contributor.totalAmountDonated) {
+      continue;
+    }
+
+    // Put contributors in the array corresponding to their types
+    if (contributor.type === CollectiveType.USER) {
+      topIndividuals.push(contributor);
+    } else if (contributor.type === CollectiveType.ORGANIZATION || contributor.type === CollectiveType.COLLECTIVE) {
+      topOrgs.push(contributor);
+    }
+
+    if (topIndividuals.length >= 10 && topOrgs.length >= 10) {
+      break;
+    }
+  }
+
+  // If one of the two categories is not filled, complete with more contributors from the other
+  const nbColsPerCategory = 2;
+  const nbFreeColsFromOrgs = nbColsPerCategory - Math.ceil(topOrgs.length / 5);
+  const nbFreeColsFromIndividuals = nbColsPerCategory - Math.ceil(topOrgs.length / 5);
+  let takeNbOrgs = 10;
+  let takeNbIndividuals = 10;
+
+  if (nbFreeColsFromOrgs > 0) {
+    takeNbIndividuals += nbFreeColsFromOrgs * 5;
+  } else if (nbFreeColsFromIndividuals > 0) {
+    takeNbOrgs += nbFreeColsFromIndividuals * 5;
+  }
+
+  return [topOrgs.slice(0, takeNbOrgs), topIndividuals.slice(0, takeNbIndividuals)];
+};
+
+export const getTopContributorsMemoized = memoizeOne(getTopContributors);
```

Comment:
Memoization won't work if the function is called with different arguments in different places.

Reply:
Oh interesting, I guess I've misunderstood it then.

So having a memoized function in a library won't work?

Or will it not work in this particular way?

i.e. how can I avoid copy-pasting the exact same 'top contributors' logic in the Contribute and Budget sections during this transitional period, or is that a debt we're willing to take on knowing it'll be cleaned up at the end of the new collective navbar project?